cloudinit/sources/helpers/vmware/imc/guestcust_util.py

```diff
@@ -265,6 +265,17 @@ def get_data_from_imc_raw_data_cust_cfg(cust_cfg):
             )
             return (None, None, None)
 
+        try:
+            logger.debug("Validating if meta data is valid or not")
+            md = safeyaml.load(md)
```

Comment:
This is translating a string to some other structure...likely a dict or list. This means the return type of this function will be different than before. Is that what you're expecting?

Reply:
Many thanks for the review. Yes, this is expected.  Metadata got from different transports in DS VMware will be loaded at https://github.com/canonical/cloud-init/blob/main/cloudinit/sources/DataSourceVMware.py#L626,  in this load_json_or_yaml() function, it will return data directly if it's already a dict.  Now this change is to validate the metadata from imc transport early so that we get a chance to report customization error status to vSphere, this is the same behavior when imc data transport was in DS OVF.